facebookresearch/ParlAI
# Create Base Classes for chat services

**Patch description**
This PR tries to implement the blueprint (base-class) for the Manager and Agent for the chat services and successfully update messenger to obey this inheritance.WIP for #2079 
Currently:
- Implemented base chatServiceManager
- Implement base Agent (@domrigoglioso )
TODO:
- Incorporate socket runner etc. as mentioned [here](https://github.com/facebookresearch/ParlAI/issues/2079#issuecomment-544614510)
- Update ParlAI messenger to obey this structure (Future PR)

**Testing steps**
@klshuster , could you point us to some tests that we could run to make sure things aren't broken ?

**Logs**
N/A 

**Other information**
N/A

**Data tests (if applicable)**
N/A
